fix: support adding prefixes and suffixes in Bulk Edit Memo

matveevtam opened the pull request:
GitHub Issue (if applicable): #2917

Trello Link (if applicable): https://trello.com/c/GyXyE3jo/343-edit-multiple-memo-fields-at-once

**Explanation of Bugfix/Feature/Modification:**
Added selection of what you actually want to do with the memos: replace with a new value, add prefix or add suffix.

Now `EditMemo` component has 3 steps:
1. **Step 1** (default): showing just the button "Edit Memos". When you click it, it goes to **Step 2**.
2. **Step 2**: showing 4 buttons: 
    a. "Replace memos"
    b. "Add prefix to each memo"
    c. "Add suffix to each memo"
    d. "Cancel" - goes back to **Step 1**
    When you click any of the buttons a-c, the component goes to **Step 3**.
3. **Step 3**: showing:
    - a label that reminds you what you selected on **Step 2**
    - an input
    - button "Back" - go back to **Step 2**
    - button "Save" - perform bulk memo edit in the way that you chose on **Step 2**, close the menu and go back to **Step 1**).

Below are screenshots of all the possible states.

![01 - step 1](https://user-images.githubusercontent.com/23550734/194716678-11b7663c-f51b-47cb-918d-e4cd625bb757.png)
Default state - Step 1

![02 - step 2](https://user-images.githubusercontent.com/23550734/194716677-842b2781-4b54-4d56-bbab-a53888c4a741.png)
Clicked the button

![03 - step 3 - replace](https://user-images.githubusercontent.com/23550734/194716676-6e89d00e-a318-4042-a71c-f25aeba2dd59.png)
Clicked "Replace memos"

![04 - step 3 - prefix](https://user-images.githubusercontent.com/23550734/194716674-23aa7b87-1d1f-4a81-84c0-ffea5322c179.png)
Clicked "Back" and then clicked "Add prefix to each memo"

![05 - step 3 - suffix](https://user-images.githubusercontent.com/23550734/194716672-5b661c4a-fc79-45c9-bc6a-7f253c1052ec.png)
Clicked "Back" and the clicked "Add suffix to each memo"

Previously it had only **Step 1** and **Step 3**, and the only way to bulk edit memos was to replace all the old memos with the new one you inputed. Now, depending on what you selected on **Step 2**, your input will either replace the old memos, or be prepended to each of them (without adding any whitespaces), or get appended (also without adding any whitespaces).

Below are screenshots of the component built from the branch "main".
![06 - main - step 1](https://user-images.githubusercontent.com/23550734/194716670-7c76ec19-8163-4607-a04c-cac82364159c.png)
![07 - main - step 3](https://user-images.githubusercontent.com/23550734/194716669-2225e6dd-8dc8-4d44-9dbc-5c51826f822a.png)


**Additional small fix:**
Fixed paddings of the "Edit memos" menu item.

joshmadewell commented on src/extension/features/accounts/bulk-edit-memo/index.jsx:
```diff
@@ -5,57 +5,145 @@ import { l10n } from 'toolkit/extension/utils/toolkit';
+    switch (editMode) {
+      case REPLACE_EDIT_MODE:
+        return memoInputValue;
+      case ADD_PREFIX_EDIT_MODE:
+        return memoInputValue + prevMemo;
+      case ADD_SUFFIX_EDIT_MODE:
+        return prevMemo + memoInputValue;
```
I'm thinking we should maybe add a space here for the user? it might be confusing if their prefix just gets added without the space?

matveevtam replied:
I was considering this as well.

If we don't insert a whitespace, a user can press Ctrl+z to cancel an undesired update (works with updating several transactions, I checked), then they can repeat the operation, but add a whitespace.

If we force inserting a whitespace, some user might be unhappy if they want to add a prefix/suffix without one. However, this seems very unlikely and they can raise an issue in the tracker.

The way to make everyone happy would be to add a checkbox saying "add a whitespace" that's enabled by default. Might look a bit too overloaded though.

WDYT? I personally prefer the 3rd way, but don't insist.

joshmadewell commented on src/extension/features/accounts/bulk-edit-memo/index.jsx:
```diff
@@ -5,57 +5,145 @@ import { l10n } from 'toolkit/extension/utils/toolkit';
     });
 
     controllerLookup('accounts').send('closeModal');
+    setDisplayMode(DEFAULT_DISPLAY_MODE);
```
I don't think the modal is closing anymore? this might be a different bug than from your PR might need to do `getAccountService.send('closeModal')`

matveevtam commented:
@joshmadewell Hi, I've updated the PR: added a space separator when editing memo and used `getAccountsController()`, although the modal is still not closing.
